dcos/resource_dcos_package.go

```diff
@@ -596,33 +612,37 @@ func resourceDcosPackageRead(d *schema.ResourceData, meta interface{}) error {
 	var desc *dcos.CosmosServiceDescribeV1Response
 	client := meta.(*dcos.APIClient)
 
-	// If the app_id is missing, this resource is never created. Guard against
-	// this case as early as possible.
-	vString, appIdok := d.GetOk("app_id")
-	if !appIdok {
-		log.Printf("[WARN] Missing 'app_id'. Assuming the service is not installed")
+	appId, err := dcosPackageParseID(d.Id())
+
+	if err != nil {
 		d.SetId("")
-		return nil
+		return err
 	}
-	appId := stripRootSlash(vString.(string))
```

Comment:
IMO this is still needed as `/foo` and `foo` are the same exact appId

Reply:
It still happens in dcosPackageParseID() - line 104